### PR TITLE
Enhance JSON Schema and JSON Keys plugins with improved error handling and explanations

### LINKED ISSUE #585: Enhance JSON Schema and JSON Keys plugins with better error handling and explanations
### What Happened?

Our JSON Schema and JSON Keys plugins currently lack detailed error reporting and explanations, making it difficult for users to understand why their validations are failing or succeeding. We need to improve these plugins to provide more comprehensive feedback and handle a wider range of JSON input scenarios.

### What Should Have Happened?

Proposed Changes

Update both plugins to handle JSON within code blocks and in plain text.
Enhance error reporting to include detailed explanations of validation results.
Improve JSON extraction to handle multiple JSON objects within a single response.
Update the plugins to return more informative data objects, including matched JSON, present/missing keys, and validation errors.
Refactor the test suites for both plugins to cover new functionality and edge cases.

Expected Outcome

Users will receive more detailed feedback on why their JSON validations succeed or fail.
The plugins will be more robust, handling a wider range of input scenarios.
Debugging and troubleshooting will be easier with the additional context provided.

### Relevant Code Snippet

_No response_

### Your Twitter/LinkedIn

_No response_

### PR DESCRIPTION
**Pull request by roh26it**
## Description
This PR addresses the issues raised in #[Issue Number] by enhancing our JSON Schema and JSON Keys plugins with improved error handling, more detailed explanations, and better JSON extraction capabilities.

## Changes Made
1. Updated both plugins to handle JSON within code blocks and in plain text.
2. Enhanced error reporting to include detailed explanations of validation results.
3. Improved JSON extraction to handle multiple JSON objects within a single response.
4. Updated the plugins to return more informative data objects, including:
   - Matched JSON
   - Present/missing keys (for JSON Keys plugin)
   - Validation errors (for JSON Schema plugin)
   - Explanatory messages for successful and failed validations
5. Refactored and expanded test suites for both plugins to cover new functionality and edge cases.

## How to Test
1. Run the updated test suites for both plugins.
2. Manually test the plugins with various input scenarios, including:
   - JSON within code blocks
   - Multiple JSON objects in a single response
   - Invalid JSON
   - Various validation scenarios (e.g., missing keys, schema mismatches)

## Expected Behavior
- The plugins should now provide more detailed feedback on validation results.
- JSON extraction should work correctly for various input formats.
- Error messages and explanations should be clear and informative.

## Additional Notes
This update significantly improves the usability and robustness of our JSON validation plugins. Users will now have a better understanding of why their validations succeed or fail, making debugging and troubleshooting much easier.

## Checklist
- [x] Code has been tested thoroughly
- [x] Documentation has been updated to reflect changes
- [x] All new and existing tests pass
- [x] Code follows project coding standards and guidelines
 
Fixes #585

### CODE REVIEW

File: plugins/default/default.test.ts [hunk 1]
```diff
@@ -45,39 +89,63 @@ describe('jsonSchema handler', () => {
...
+    const result = await jsonSchemaHandler(context, parameters, eventType);
 
+    expect(result.error).toBe(null);
+    expect(result.verdict).toBe(false);
+    expect(result.data).toBeDefined();
+    expect(result.data.explanation).toContain('No valid JSON found');
+  });
+});
+
+describe.only('jsonKeys handler', () => {
```
Comment (VisargD) [inline plugins/default/default.test.ts]: Hey @roh26it - Can you please remove `.only` from this describe statement?

Comment (VisargD) [reply plugins/default/default.test.ts]: Pushed this

**Review by VisargD — APPROVED** [review]
